src/openfermion/ops/_binary_operator.py

```diff
@@ -0,0 +1,494 @@
+# symbolic binary class for decoder definitions (arxiv 1712.07067)
+
+import copy
+import numpy as np
```

Comment:
okay, here it is, perhaps the most despised rule of OpenFermion: we have a strict "no importing as" rule. I know it's kind of silly in this case since np is almost universally recognized as numpy. But at this point we should stay consistent with other parts of the code.